Insertable Streams for MediaStreamTrack API (#8260)
mdn/content: files/en-us/web/api/mediastreamtrackgenerator/writable/index.md

---
title: MediaStreamTrackGenerator.writable
slug: Web/API/MediaStreamTrackGenerator/writable
tags:
  - API
  - Property
  - Reference
  - writable
  - MediaStreamTrackGenerator
browser-compat: api.MediaStreamTrackGenerator.writable
---
{{DefaultAPISidebar("Insertable Streams for MediaStreamTrack API")}}

> The **`writable`**  property of the {{domxref("MediaStreamTrackGenerator")}} interface returns a {{domxref("WritableStream")}}. This allows the writing of media frames to the `MediaStreamTrackGenerator`. The frames will be audio or video, the type is dictated by the kind of `MediaStreamTrackGenerator` that has been created.

Reviewer: ```suggestion
The **`writable`**  property of the {{domxref("MediaStreamTrackGenerator")}} interface returns a {{domxref("WritableStream")}}. This allows the writing of media frames to the `MediaStreamTrackGenerator`. The frames will be audio or video. The type is dictated by the kind of `MediaStreamTrackGenerator` that was created.
```
